torch/_dynamo/variables/misc.py

```diff
@@ -224,8 +224,16 @@ def visit(node):
         VariableTracker.apply(visit, (args, kwargs))
 
         if requires_grad and torch.is_grad_enabled():
-            # TODO(jansel): handle this in training mode
-            unimplemented("autograd.Function with requires_grad")
+            from .torch import TorchHigherOrderOperator
+
+            def trampoline_autograd_fn(*args, **kwargs):
```

Comment:
This is basically a bad version of CustomFunctionHigherOrderOperator - I did not realize CustomFunctionHigherOrderOperator existed when I wrote this (Thanks @zou3519)

Reply:
That's okay, we can change it later